stl/inc/ranges

```diff
@@ -6821,6 +6822,496 @@ namespace ranges {
+                (noexcept(_STD declval<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>&>() +=
+                          static_cast<iter_difference_t<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>>(
+                              _STD declval<difference_type>()))
+                    && ...)) // strengthened
+                requires _All_random_access<_IsConst, _ViewTypes...> {
+                _Tuple_for_each(
+                    [&_Off]<class _IteratorType>(
+                        _IteratorType& _Itr) { _Itr += static_cast<iter_difference_t<_IteratorType>>(_Off); },
+                    _Current);
+
+                return *this;
+            }
+
+            constexpr _Iterator& operator-=(const difference_type _Off) noexcept(
+                (noexcept(_STD declval<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>&>() -=
+                          static_cast<iter_difference_t<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>>(
+                              _STD declval<difference_type>()))
+                    && ...)) // strengthened
+                requires _All_random_access<_IsConst, _ViewTypes...> {
+                _Tuple_for_each(
+                    [&_Off]<class _IteratorType>(
+                        _IteratorType& _Itr) { _Itr -= static_cast<iter_difference_t<_IteratorType>>(_Off); },
+                    _Current);
+
+                return *this;
+            }
+
+            _NODISCARD constexpr auto operator[](const difference_type _Where) const
+                noexcept((noexcept((_STD declval<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>())
+                                  [static_cast<iter_difference_t<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>>(
+                                      _STD declval<difference_type>())])
+                          && ...)) // strengthened
+                requires _All_random_access<_IsConst, _ViewTypes...> {
+                return _Tuple_transform(
+                    [&_Where]<class _IteratorType>(_IteratorType& _Itr) -> decltype(auto) {
+                        return _Itr[static_cast<iter_difference_t<_IteratorType>>(_Where)];
+                    },
+                    _Current);
+            }
+
+            _NODISCARD_FRIEND constexpr bool operator==(const _Iterator& _Lhs, const _Iterator& _Rhs) noexcept(
+                noexcept(_STD declval<_Iterator_tuple_type>() == _STD declval<_Iterator_tuple_type>())) // strengthened
+                requires(equality_comparable<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>&&...) {
+                if constexpr (!_Zip_is_common<_Maybe_const<_IsConst,
+                                  _ViewTypes>...> || _All_random_access<_IsConst, _ViewTypes...>) {
+                    return (_STD get<0>(_Lhs._Current) == _STD get<0>(_Rhs._Current));
+                } else if constexpr (_All_bidirectional<_IsConst, _ViewTypes...>) {
+                    return (_Lhs._Current == _Rhs._Current);

```

Comment:
Just to clarify, you meant `!_Zip_is_common` in that branch condition, right? Otherwise, if `_Zip_is_common` is true, then `zip_view::end()` might return an `_Iterator` containing end iterators which are never used to check for the end of the view.